Controller/Payment/Action.php

```diff
@@ -0,0 +1,356 @@
+        $this->_creditmemoRepository = $creditmemoRepository;
+        $this->_midtransLogger = $midtransLogger;
+        $this->registry = $registry;
+        $this->_customerSession = $customerSession;
+        $this->_contextHttp = $contextHttp;
+        $this->context = $context;
+        $this->resultFactory = $context->getResultFactory();
+        $this->_redirect = $context->getRedirect();
+        $this->resultRedirectFactory = $context->getResultRedirectFactory();
+        $this->_request = $context->getRequest();
+        $this->_response = $context->getResponse();
+        $this->paymentOrderRepository = $paymentOrderRepository;
+        $this->paymentRequestRepository = $paymentRequestRepository;
+    }
+
+    /**
+     * Retrieve request object
+     *
+     * @return RequestInterface
+     */
+    public function getRequest()
+    {
+        return $this->_request;
+    }
+
+    /**
+     * Retrieve response object
+     *
+     * @return ResponseInterface
+     */
+    public function getResponse()
+    {
+        return $this->_response;
+    }
+
+    /**
+     * Set redirect into response
+     *
+     * @param string $path
+     * @param array $arguments
+     * @return ResponseInterface
+     */
+    protected function _redirect($path, $arguments = [])
+    {
+        $this->_redirect->redirect($this->getResponse(), $path, $arguments);
+        return $this->getResponse();
+    }
+
+    /**
+     * Get increment id from last order checkout session
+     *
+     * @return string
+     */
+    public function getOrderIdSession()
+    {
+        return $this->_checkoutSession->getLastRealOrder()->getIncrementId();
+    }
+
+    /**
+     * Get order by incrementId from session checkout
+     *
+     * @return Order
+     */
+    public function getOrderFromSession()
+    {
+        return $this->_order->loadByIncrementId($this->getOrderIdSession());
+    }
+
+    /**
+     * Get Midtrans data config
+     *
+     * @return Data
+     */
+    public function getMidtransDataConfig()
+    {
+        return $this->data;
+    }
+
+    /**
+     * Set value to core session
+     *
+     * @param $order_id
+     */
+    public function setValue($order_id)
+    {
+        $this->_coreSession->start();
+        $this->_coreSession->setMessage($order_id);
+    }
+
+    /**
+     * get value from core session
+     *
+     * @return mixed
+     */
+    public function getValue()
+    {
+        $this->_coreSession->start();
+        return $this->_coreSession->getMessage();
+    }
+
+    /**
+     * unset value from core sessions
+     *
+     * @return mixed
+     */
+    public function unSetValue()
+    {
+        $this->_coreSession->start();
+        return $this->_coreSession->unsMessage();
+    }
+
+    /**
+     * Get Midtrans status via API
+     *
+     * @param $param
+     * @param null $code
+     * @return mixed[]
+     * @throws \Exception
+     */
+    public function midtransGetStatus($param, $code = null)
```

Comment:
@rizdaprasetya param it can be `Midtrans order/trx-id` or `Magento order objec`t, that’s it. agree it's not descriptive. Any idea? can I used $midtransOrderIdorMagentoOrder or put the descriptive on PHP docs.

Yes the `$code` it should be `$paymentCode` sure will change this

Reply:
Done, updated the PHP docs